lib/python/qmk/constants.py

```diff
@@ -17,6 +17,45 @@
 LUFA_PROCESSORS = 'at90usb162', 'atmega16u2', 'atmega32u2', 'atmega16u4', 'atmega32u4', 'at90usb646', 'at90usb647', 'at90usb1286', 'at90usb1287', None
 VUSB_PROCESSORS = 'atmega32a', 'atmega328p', 'atmega328', 'attiny85'
 
+# Bootloaders of the supported processors
+MCU2BOOTLOADER = {
```

Comment:
There is not exactly a 1:1 mapping here, some MCUs can support multiple bootloaders. May be outside the scope of this PR, but I think the value here should be lists of valid bootloaders, or strings if only one is valid (other than `custom`).
eg.
```python
"STM32F103": "stm32duino",
# ...
"atmega32a": ["bootloadhid", "usbasploader"],
```

Reply:
Yeah, that's a good point.
I think we can get back to this in a follow-up PR.